src/Tizen.NUI/src/public/VisualMaps.cs

```diff
@@ -660,27 +660,85 @@ protected virtual void ComposingPropertyMap()
+                _outputVisualMap.Add(Visual.Property.Shader, temp);
+                temp.Dispose();
+            }
+            if (_premultipliedAlpha != null)
+            {
+                PropertyValue temp = new PropertyValue((bool)_premultipliedAlpha);
+                _outputVisualMap.Add(Visual.Property.PremultipliedAlpha, temp);
+                temp.Dispose();
+            }
+            if (_mixColor != null)
+            {
+                PropertyValue temp = new PropertyValue(_mixColor);
+                _outputVisualMap.Add(Visual.Property.MixColor, temp);
+                temp.Dispose();
+            }
+            if (_opacity != null)
+            {
+                PropertyValue temp = new PropertyValue((float)_opacity);
+                _outputVisualMap.Add(Visual.Property.Opacity, temp);
+                temp.Dispose();
+            }
+            if (_visualFittingMode != null)
+            {
+                PropertyValue temp = new PropertyValue((int)_visualFittingMode);
+                _outputVisualMap.Add(Visual.Property.VisualFittingMode, temp);
+                temp.Dispose();
+            }
+            if (cornerRadius != null)
+            {
+                PropertyValue temp = new PropertyValue((int)cornerRadius);
+                _outputVisualMap.Add(Visual.Property.CornerRadius, temp);
+                temp.Dispose();
+            }
         }
 
         private void ComposingTransformMap()
         {
-            _visualTransformMap = new PropertyMap();
-            if (_visualSize != null) { _visualTransformMap.Add((int)VisualTransformPropertyType.Size, new PropertyValue(_visualSize)); }
-            if (_visualOffset != null) { _visualTransformMap.Add((int)VisualTransformPropertyType.Offset, new PropertyValue(_visualOffset)); }
-            if (_visualOffsetPolicy != null) { _visualTransformMap.Add((int)VisualTransformPropertyType.OffsetPolicy, new PropertyValue(_visualOffsetPolicy)); }
-            if (_visualSizePolicy != null) { _visualTransformMap.Add((int)VisualTransformPropertyType.SizePolicy, new PropertyValue(_visualSizePolicy)); }
-            if (_visualOrigin != null) { _visualTransformMap.Add((int)VisualTransformPropertyType.Origin, new PropertyValue((int)_visualOrigin)); }
-            if (_visualAnchorPoint != null) { _visualTransformMap.Add((int)VisualTransformPropertyType.AnchorPoint, new PropertyValue((int)_visualAnchorPoint)); }
+            visualTransformMap = new PropertyMap();
+            if (visualSize != null)
+            {
+                PropertyValue temp = new PropertyValue(visualSize);
+                visualTransformMap.Add((int)VisualTransformPropertyType.Size, temp);
+                temp.Dispose();
+            }
+            if (visualOffset != null)
+            {
+                PropertyValue temp = new PropertyValue(visualOffset);
+                visualTransformMap.Add((int)VisualTransformPropertyType.Offset, temp);
```

Comment:
could you check if the `temp` is copied when `.Add()` is called?
I am concerned about that `visualTransfromMap` is used in other place and if `temp` is disposed right after `Add()`, then the `visualTransformMap` could have null `ProperyValue` inside. 
how do you think about?

Reply:
> could you check if the `temp` is copied when `.Add()` is called?
> I am concerned about that `visualTransfromMap` is used in other place and if `temp` is disposed right after `Add()`, then the `visualTransformMap` could have null `ProperyValue` inside.
> how do you think about?
I have checked the Dali codes:
 Dali::Property::Value *arg3 = 0 ;
result = (Dali::Property::Map *) &(arg1)->Add(arg2,(Dali::Property::Value const &)*arg3);
The arg3 is the pointer of ProperyValue, it did value copied when Adding. So it is safe as Ms.Yang said.